build-logic/convention/src/main/kotlin/com/google/samples/apps/nowinandroid/AndroidCompose.kt

```diff
@@ -52,11 +54,24 @@ internal fun Project.configureAndroidCompose(
 
     extensions.configure<ComposeCompilerGradlePluginExtension> {
         if (isPropertyValueIsTrue("enableComposeCompilerReportsAndMetrics")) {
-            metricsDestination = relativeToRootProject("compose-metrics")
-            reportsDestination = relativeToRootProject("compose-reports")
+            metricsDestination = createPerModulePath("compose-metrics")
+            reportsDestination = createPerModulePath("compose-reports")
         }
 
-        stabilityConfigurationFile =
-            rootProject.layout.projectDirectory.file("compose_compiler_config.conf")
+        stabilityConfigurationFiles
+            .add(isolated.rootProject.projectDirectory.file("compose_compiler_config.conf"))
     }
 }
+
+/**
+ * Checking a value of [propertyName] is true.
+ */
+private fun Project.isPropertyValueIsTrue(propertyName: String): Boolean =
+    providers.gradleProperty(propertyName).orNull.toBoolean()
+
+/**
+ * Creating a path where the [dir] will be created in.
+ */
+private fun Project.createPerModulePath(dir: String): Provider<Directory> = provider {
```

Comment:
Again, this is not "creating" a directory.
And when I said "declared as an inlined extension", here is what I meant:

```kotlin
val isComposeCompilerReportsAndMetricsEnabled = providers.gradleProperty("enableComposeCompilerReportsAndMetrics").orNull.toBoolean()
if (isComposeCompilerReportsAndMetricsEnabled) {
    fun relativeToRootProject(dir: String) = isolated.rootProject.projectDirectory.dir("build").dir(projectDir.toRelativeString(rootDir)).dir(dir)
    metricsDestination = relativeToRootProject("compose-metrics")
    reportsDestination = relativeToRootProject("compose-reports")
}
```

Reply:
I replaced it in a inlined function.